Fix: Handle undefined release.body in release.js to prevent replace()…

sharvitKashikar opened the pull request:
## Summary
- Fixed a bug where `release.body.replace()` caused a TypeError if `release.body` was undefined.
- Now checks `release && release.body` before calling `.replace()`.
- Added a fallback `"No release notes available."` for missing release notes.
- Ensured `.split()` does not fail when extracting release notes.



## Testing
- Tested locally on `localhost:8080/release/v27.0.0`
- Page loads successfully without errors
- Handles missing release notes gracefully

codebytere commented on src/routes/release.js:
```diff
@@ -197,15 +197,15 @@ router.get(
-    releaseNotes =
-      '# Release Notes\n' +
-      releaseNotes.replace(/# Release Notes for [^\r\n]+(?:(?:\n)|(?:\r\n))/i, '');
+    let releaseNotes = release && release.body ? release.body : 'No release notes available.';
```
```suggestion
    let releaseNotes = release?.body ?? 'No release notes available.';
```

codebytere commented on src/routes/release.js:
```diff
@@ -249,4 +249,4 @@ router.get(
 );
 
-module.exports = router;
+module.exports = router;
```
Newline should not have been removed.

codebytere commented on src/routes/release.js:
```diff
@@ -197,15 +197,15 @@ router.get(
+
+const parsed = semver.parse(version);
+if (parsed.prerelease.length) {
+  releaseNotes = releaseNotes.split(new RegExp(`@${escapeRegExp(version.substr(1))}\`?.`))[1] || releaseNotes;
+}
```
If we already now know `releaseNotes` will be a valid string, we don't need `|| releaseNotes;`

codebytere commented on src/routes/release.js:
```diff
@@ -197,15 +197,15 @@ router.get(
+}
 
+//  replace() now only runs if releaseNotes is valid
+releaseNotes = '# Release Notes\n' + (releaseNotes ? releaseNotes.replace(/# Release Notes for [^\r\n]+(?:(?:\n)|(?:\r\n))/i, '') : '');
```
We know `releaseNotes` will be defined as of this point so we don't need this check here either.

sharvitKashikar commented:
Hey @codebytere, I’ve addressed all the requested changes. Let me know if there’s anything else needed. Thanks for your time!

erickzhao commented on src/routes/release.js:
```diff
@@ -20,7 +20,7 @@ const router = new Router();
 Handlebars.registerHelper('ifEquals', function (arg1, arg2, options) {
   return arg1 == arg2 ? options.fn(this) : options.inverse(this);
-});
+});a
```
typo here

sharvitKashikar replied:
@erickzhao I am really sorry that I did not see it earlier. 
Thanks for letting me know. I have addressed this in the latest push

erickzhao commented on src/routes/release.js:
```diff
@@ -197,15 +197,14 @@ router.get(
-    releaseNotes =
-      '# Release Notes\n' +
-      releaseNotes.replace(/# Release Notes for [^\r\n]+(?:(?:\n)|(?:\r\n))/i, '');
+ let releaseNotes = release?.body ?? 'No release notes available.';
```
Did you run the `prettier:write` command? This uneven whitespace seems like it would fail.

sharvitKashikar commented:
@erickzhao I addressed all the requested changes in the latest push. 
- removed the typo 
- ran `prettier:write` command to remove the uneven spaces

sharvitKashikar commented:
Hey @codebytere @erickzhao, just checking in! I've made all the requested changes. 
Let me know if anything else is needed or if I should update something else. 
Thanks for your time!

erickzhao reviewed (approved):
Pushed up a few commits to just minimize the diff but otherwise LGTM.

codebytere commented:
@sharvitKashikar commits must have verified signatures - please correct and then we can proceed merge.

erickzhao commented:
Hi @sharvitKashikar, your commits are still unverified after your last force push.

dsanders11 commented:
@sharvitKashikar, I'm going to close this for now as we require verified commits and can't accept this without them. If you clean it up and want to get it merged ping me and I'll reopen it. 👍

sharvitKashikar commented:
> @sharvitKashikar, I'm going to close this for now as we require verified commits and can't accept this without them. If you clean it up and want to get it merged ping me and I'll reopen it. 👍

@dsanders11  I can work on this, I can now sign and verify my commits,  I can clean it and work on it
Thank you
